### DISCUSSION TITLE
Binary classification using roll_time_series

### DISCUSSION THREAD

**Discussion by foooooooooooooooobar** [blue-yonder/tsfresh#841] [Q&A] [answered]
Hopefully quick question about usage of roll_time_series with usage in stock data.

Let's say I am just using the "close" column (which is the closing price of a stock) to make the question easier.

I am using it as follows:

1. Call roll_time_series on the dataframe
1. Call extract_features on this data
1. Create lags of these values (so i.e. I would have close_energy_ratio_t_1) where _t_1 indicates it is the value of the previous row (date sorted)
1. Pass these to a binary classification task, which predicts if the stock goes "up" (whether stock jumped more than 5%).

My questions are this:
- Will any of those data leak info from future rows? (i.e. is close_energy using future data to get that stat)? Looking at the docs https://tsfresh.readthedocs.io/en/latest/text/forecasting.html I think all the features are calculated on data up to that window, so if I access the previous row's value I think I should be fine?

Thanks

**Comment by nils-braun** [comment]
Sorry for the late answer :-(
I think I did not 100% understand your 3rd step (why do you need to lag again? The rolling is already creating windows, but maybe I got it wrong).

But in principle you are correct: all feature calculators are always only computer in the data of a single ID and value. After the rolling, all data of a single window have the same id. So you should be fine, yes.
If you don't "trust" your pipeline, you can also use very simple data (1,2,3,...) And just a bunch of simple calculators (mean, sum), so that you can actually see what is going on

**Comment by foooooooooooooooobar** [comment, marked as answer]
No problem at all! I greatly appreciate your work on an open-source library. I guess there are two ways to do it.

If you have data like
date            | close | close median
01/01/2012 |   10    | 10

And if you want to make predictions on whether the stock will increase 5%, **without lag** columns, you would say that those features "date, close, and close median" for 01/01 would predict if **tomorrow's** price would jump or not.

**With** lag columns, so you're features you pass to the predition are `close_t_1` and `close_median_t_1` you would predict if 01/01 would jump or not. I chose the latter because I also track `open` data so it was easy for them to be together, but really it's just semantics.

**Comment by nils-braun** [comment]
I see, now it makes sense! Yes, looks good. I guess you have already seen our stock prediction notebooks on that? If you think it makes sense to add your method as another example feel free to do a PR :-)
(And if you are getting rich with stock market predictions, tell us ;-))
